describe/LN.ts

```diff
@@ -0,0 +1,118 @@
+import { AbstractDataAttributeDescription } from "./AbstractDataAttribute.js";
+import { LNodeType, LNodeTypeDescription } from "./LNodeType.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { describeVal, sortValues } from "./Val.js";
+
+export interface LNDescription extends NamingDescription {
+  lnType: LNodeTypeDescription;
+}
+
+/** Returns leaf data attribute (BDA or DA) from
+ * LNodeTypeDescription containing vals
+ * @param path - parent DOI/SDI/DAI name attributes
+ * @param lNodeType - LNodeTypeDescription of the logical node
+ * */
+function getLeafDataAttribute(
+  path: string[],
+  lNodeType: LNodeTypeDescription
+): Object {
+  let index = 0;
+  let dataType: any = lNodeType;
+
+  let leafData = {};
+  while (path.length > index) {
+    if (dataType.dos) {
+      // LNodeType->DO
+      const newData = dataType.dos[path[index]].type;
+      dataType = newData;
+    } else if (dataType.sdos || dataType.das) {
+      // DOType
+      if (dataType.sdos[path[index]]) {
+        //SDO
+        const sdo = dataType.sdos[path[index]];
+        const newData = sdo.type;
+        dataType = newData;
+      } else if (dataType.das[path[index]]) {
+        //DA
+        const da = dataType.das[path[index]];
+        if (da.bType === "Struct") {
+          const newData = da.type;
+          dataType = newData;
+        } else {
+          leafData = da;
+        }
+      }
+    } else if (dataType.bdas) {
+      // DAType->BDA
+      const bda = dataType.bdas[path[index]];
+      if (bda.bType === "Struct") {
+        const newData = bda.type;
+        dataType = newData;
+      } else {
+        leafData = bda;
+      }
+    }
+    index++;
+  }
+
+  return leafData;
+}
+
+function updateValues(
+  lNodeType: LNodeTypeDescription,
+  instanceValues: Record<string, Element[]>
+): LNodeTypeDescription {
+  Object.entries(instanceValues).forEach(([key, value]) => {
+    const leafDataAttribute = getLeafDataAttribute(
+      key.split("."),
+      lNodeType
+    ) as AbstractDataAttributeDescription;
+
+    leafDataAttribute.vals = value
+      .map((val) => describeVal(val))
+      .sort(sortValues);
+  });
+
+  return lNodeType;
+}
+
+/** Path consisting of parents name attributes: e.g. ["Beh","stVal"] */
+function pathToInstanceValue(element: Element): string[] | undefined {
+  if (element.tagName === "LN" || element.tagName === "LN0") return [];
+
+  const parent = element.parentElement!;
+
+  if (element.tagName === "Val") return pathToInstanceValue(parent);
+
+  const parentPath = pathToInstanceValue(parent);
```

Comment:
I'd find this more readable:

```suggestion
  const parentPath = pathToInstanceValue(element.parentElement!);

  if (element.tagName === "Val") return parentPath;
```